core/spec/models/spree/promotion_handler/free_shipping_spec.rb

```diff
@@ -13,8 +13,20 @@ module PromotionHandler
       context 'with apply_automatically' do
         let!(:promotion) { create(:promotion, apply_automatically: true, promotion_actions: [action]) }
 
-        it "creates the adjustment" do
-          expect { subject.activate }.to change { shipment.adjustments.count }.by(1)
+        context 'for eligible promotion' do
+          before { allow_any_instance_of(Spree::Promotion).to receive(:eligible?).and_return(true) }
+
+          it "creates the adjustment" do
+            expect { subject.activate }.to change { shipment.adjustments.count }.by(1)
+          end
+        end
+
+        context 'for ineligible promotion' do
+          before { allow_any_instance_of(Spree::Promotion).to receive(:eligible?).and_return(false) }
```

Comment:
Rather than stubbing would you mind instead creating a promotion that will in fact not be eligible?  e.g. 
```ruby
let(:promotion) { create(:promotion, :with_item_total_rule, item_total_threshold_amount: 1_000, ... }
```